### PR TITLE
PMM-4310 pmm-agent's systemd unit should restart it on crash

### PR DESCRIPTION
**Pull request by EvgeniyPatlan**

### CODE REVIEW

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-332-e7f9b97
client docker - perconalab/pmm-client-fb:PR-332-e7f9b97
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-332-e7f9b97.tar.gz

File: build/config/pmm-agent.service [hunk 1]
```diff
@@ -6,6 +6,8 @@ After=time-sync.target network.target
 Type=simple
 PermissionsStartOnly=true
 ExecStart=/usr/sbin/pmm-agent --config-file=/usr/local/percona/pmm-agent.yaml
+Restart=on-failure
```
Comment (AlekSi) [inline build/config/pmm-agent.service]: Why not `always`?

File: build/config/pmm-agent.service [hunk 2]
```diff
@@ -6,6 +6,8 @@ After=time-sync.target network.target
...
+RestartPreventExitStatus=1
```
Comment (AlekSi) [inline build/config/pmm-agent.service]: Why? We should still restart it even with exit status 1.

I think we should also add
```
RestartSec=2s
TimeoutStopSec=10s
```

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-332-8cbcb8b
client docker - perconalab/pmm-client-fb:PR-332-8cbcb8b
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-332-8cbcb8b.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-332-6545156
client docker - perconalab/pmm-client-fb:PR-332-6545156
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-332-6545156.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-332-a05a027
client docker - perconalab/pmm-client-fb:PR-332-a05a027
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-332-a05a027.tar.gz

**Pull request merged**